src/python/foglamp/core/scheduler_db_services.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""Storage Services as needed by Processes
+"""
+
+import asyncpg
+
+__author__ = "Amarendra Kumar Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+__DSN = "postgresql://foglamp:foglamp@localhost:5432/foglamp"
+__DB_NAME = 'foglamp'
+
+
+async def create_schedule(payload):
+    pass
+
+async def read_schedule(schedule_id=None):
+    conn = await asyncpg.connect(dsn=__DSN)
+    query = '''
+        SELECT id::"varchar",
+                process_name,
+                schedule_name,
+                schedule_type,
+                schedule_interval::"varchar",
+                schedule_time::"varchar",
+                schedule_day::"varchar",
+                exclusive
+        from schedules
+    '''
+
+    # TODO: Investigate why prepared statement is not working
+    # if schedule_id is None:
+    #     _where_clause = ''
+    # else:
+    #     _where_clause = ' WHERE id = \'$1\''
+    #
+    # query += _where_clause
+    #
+    # stmt = await conn.prepare(query)
+    # if schedule_id is None:
+    #     rows = await stmt.fetch()
+    # else:
+    #     rows = await stmt.fetch(schedule_id)
+
+    if not schedule_id:
+        _where_clause = ''
+    else:
+        _where_clause = ' WHERE id = \'{0}\''
```

Comment:
Putting the {0} here but the .format on the query seems a little risky to me. It is prone to error if the code is modified and new parameters introduced. I would prefer the format and placeholder to be together.